plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftMetastoreModule.java

```diff
@@ -42,16 +48,33 @@ public class ThriftMetastoreModule
     @Override
     protected void setup(Binder binder)
     {
-        OptionalBinder.newOptionalBinder(binder, ThriftMetastoreClientFactory.class)
-                .setDefault().to(DefaultThriftMetastoreClientFactory.class).in(Scopes.SINGLETON);
-        binder.bind(TokenAwareMetastoreClientFactory.class).to(StaticTokenAwareMetastoreClientFactory.class).in(Scopes.SINGLETON);
-        configBinder(binder).bindConfig(StaticMetastoreConfig.class);
-        configBinder(binder).bindConfig(ThriftMetastoreConfig.class);
+        StaticMetastoreConfig staticMetastoreConfig = buildConfigObject(StaticMetastoreConfig.class);
+        requireNonNull(staticMetastoreConfig.getMetastoreUris(), "metastoreUris is null");
+        boolean hasHttpMetastore = staticMetastoreConfig.getMetastoreUris().stream()
+                .anyMatch(uri -> "http".equalsIgnoreCase(uri.getScheme()));
+        boolean hasHttpsMetastore = staticMetastoreConfig.getMetastoreUris().stream()
+                .anyMatch(uri -> "https".equalsIgnoreCase(uri.getScheme()));
+
+        if (hasHttpMetastore || hasHttpsMetastore) {
+            OptionalBinder.newOptionalBinder(binder, ThriftMetastoreClientFactory.class)
+                    .setDefault().to(HttpThriftMetastoreClientFactory.class).in(Scopes.SINGLETON);
+            binder.bind(IdentityAwareMetastoreClientFactory.class).to(StaticTokenAwareHttpMetastoreClientFactory.class).in(Scopes.SINGLETON);
+            configBinder(binder).bindConfig(StaticMetastoreConfig.class);
+            configBinder(binder).bindConfig(ThriftHttpMetastoreConfig.class);
+            configBinder(binder).bindConfig(ThriftMetastoreConfig.class);
+        }
+        else {
+            OptionalBinder.newOptionalBinder(binder, ThriftMetastoreClientFactory.class)
+                    .setDefault().to(DefaultThriftMetastoreClientFactory.class).in(Scopes.SINGLETON);
+            binder.bind(TokenAwareMetastoreClientFactory.class).to(StaticTokenAwareMetastoreClientFactory.class).in(Scopes.SINGLETON);
+            configBinder(binder).bindConfig(StaticMetastoreConfig.class);
+            configBinder(binder).bindConfig(ThriftMetastoreConfig.class);
+            install(new ThriftMetastoreAuthenticationModule());
+        }
 
         binder.bind(ThriftMetastoreFactory.class).to(ThriftHiveMetastoreFactory.class).in(Scopes.SINGLETON);
```

Comment:
this factory is related to thrift only usecases

you'll probably need a dedicated simplified class for http/https usecases (which does not do deletefiles on drop,  use spark statistics fallback ,etc.)

Reply:
I see. Do you think we need to create a new ThriftHttpMetastore class as well?